geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/BinAggregatingIterator.scala

```diff
@@ -72,32 +72,54 @@ class PrecomputedBinAggregatingIterator extends BinAggregatingIterator {
     // we only need to decode the parts required for the filter/dedupe/sampling check
     // note: we wouldn't be using precomputed if sample by field wasn't the track id
     decodeBin = if (filter) {
-      setDate = if (isDtgArray) {
+      val dtgIndex = encoding.dtgField.get
+      val geomIndex = encoding.geomField.get
+      val trackIndex = encoding.trackIdField.get
+      setDate = if (sft.getDescriptor(dtgIndex).getType.getBinding == classOf[Date]) {
+        (sf, long) => sf.setAttribute(dtgIndex, new Date(long))
+      } else {
         (sf, long) => {
           val list = new java.util.ArrayList[Date](1)
           list.add(new Date(long))
           sf.setAttribute(dtgIndex, list)
         }
-      } else {
-        (sf, long) => sf.setAttribute(dtgIndex, new Date(long))
       }
-      (_) => {
-        val row = source.getTopKey.getRow
-        sf.setId(getId(row.getBytes, 0, row.getLength))
-        setValuesFromBin(sf, gf)
-        sf
+
+      val callback = new BinaryOutputCallback() {
+        override def apply(trackId: Int, lat: Float, lon: Float, dtg: Long): Unit = {
+          sf.setAttribute(geomIndex, gf.createPoint(new Coordinate(lat, lon)))
+          sf.setAttribute(trackIndex, Int.box(trackId)) // TODO setAttributeNoConvert
+          setDate(sf, dtg)
+        }
+        override def apply(trackId: Int, lat: Float, lon: Float, dtg: Long, label: Long): Unit =
+          apply(trackId, lat, lon, dtg)
```

Comment:
yeah, this class only operates on the precomputed bin values, which only support the non-label value... maybe i will just make this a throws since it would be an error case if it was ever called